exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExporterWorker.java

```diff
@@ -128,40 +152,60 @@ void export() {
         viewDataList.add(viewManager.getView(view.getName()));
       }
     }
+
     List<TimeSeries> timeSeriesList = Lists.newArrayList();
     for (ViewData viewData : viewDataList) {
       timeSeriesList.addAll(
           StackdriverExportUtils.createTimeSeriesList(viewData, monitoredResource));
     }
-
     for (List<TimeSeries> batchedTimeSeries :
         Lists.partition(timeSeriesList, MAX_BATCH_EXPORT_SIZE)) {
-      // Batch export 3 TimeSeries at one call, to avoid exceeding RPC header size limit.
-      CreateTimeSeriesRequest request =
-          CreateTimeSeriesRequest.newBuilder()
-              .setNameWithProjectName(projectName)
-              .addAllTimeSeries(batchedTimeSeries)
-              .build();
-      try {
+      Span span = tracer.getCurrentSpan();
+      span.addAnnotation("Export Stackdriver TimeSeries.");
+      try (Scope scope = tracer.withSpan(span)) {
+        // Batch export 3 TimeSeries at one call, to avoid exceeding RPC header size limit.
+        CreateTimeSeriesRequest request =
+            CreateTimeSeriesRequest.newBuilder()
+                .setNameWithProjectName(projectName)
+                .addAllTimeSeries(batchedTimeSeries)
+                .build();
         metricServiceClient.createTimeSeries(request);
+        span.addAnnotation("Finish exporting TimeSeries.");
+      } catch (ApiException e) {
+        logger.log(Level.WARNING, "ApiException thrown when exporting TimeSeries.", e);
+        span.setStatus(
+            Status.CanonicalCode.valueOf(e.getStatusCode().getCode().name())
+                .toStatus()
+                .withDescription(
+                    "ApiException thrown when exporting TimeSeries: " + exceptionMessage(e)));
       } catch (Throwable e) {
         logger.log(Level.WARNING, "Exception thrown when exporting TimeSeries.", e);
+        span.setStatus(
+            Status.UNKNOWN.withDescription(
+                "Exception thrown when exporting TimeSeries: " + exceptionMessage(e)));
       }
     }
   }
 
   @Override
   public void run() {
     while (true) {
-      try {
+      Span span = tracer.spanBuilder("ExportStatsToStackdriverMonitoring").startSpan();
```

Comment:
You may also use setRecordEvents(true) to track all operations and have them on the "/tracez" page? To have it on the tracez page you also need to call in the constructor of this class:

```
Tracing.getExportComponent().getSampledSpanStore().registerSpanNamesForCollection(
        Collections.singletonList("ExportStatsToStackdriverMonitoring"));
```

After you do this change you can run the ZPagesTester (in examples) with the SD stats exporter and see the spans on "localhost:8080/tracez"

Reply:
Done.